cpp/tests/groupby/rank_scan_tests.cpp

```diff
@@ -23,17 +23,25 @@
 
 #include <cudf/detail/aggregation/aggregation.hpp>
 
-using namespace cudf::test::iterators;
-
 namespace cudf {
 namespace test {
 
-inline void test_pair_rank_scans(column_view const& keys,
-                                 column_view const& order,
-                                 column_view const& expected_dense,
-                                 column_view const& expected_rank,
-                                 null_policy include_null_keys = null_policy::INCLUDE,
-                                 sorted keys_are_sorted        = sorted::YES)
+using namespace iterators;
+
+template <typename T>
+using input              = fixed_width_column_wrapper<T>;
+using rank_result_col    = fixed_width_column_wrapper<size_type>;
+using percent_result_col = fixed_width_column_wrapper<double>;
+using null_iter_t        = decltype(nulls_at({}));
+
+auto constexpr X     = int32_t{-1};  // Placeholder for NULL rows.
```

Comment:
Don't use a negative value for placeholder. Sometimes it may be cast to an unsigned type and may be out of bound if further casting to a smaller type.

Reply:
Force of habit. Fixed now.